pkgs/applications/version-management/sublime-merge/common.nix

```
@@ -53,6 +53,16 @@ in let
     installPhase = ''
       runHook preInstall
 
+      # We need to replace the ssh-askpass-sublime executable because the default one
+      # will not function properly, in order to work it needs to pass an argv[0] to
+      # the sublime_merge binary, and the built-in version will will try to call the
+      # sublime_merge wrapper script which cannot pass through the original argv[0] to
+      # the sublime_merge binary. Thankfully the ssh-askpass-sublime functionality is
+      # very simple and can be replaced with a simple bash script.
+      rm ssh-askpass-sublime
+      echo -e "#!${stdenv.shell}\nexec -a /ssh-askpass-sublime $out/.sublime_merge-wrapped" > ssh-askpass-sublime
```

You should be able to use `makeWrapper` here (untested):

```suggestion
  makeWrapper $out/.sublime_merge-wrapped $out/ssh-askpass-sublime \
    --argv0 "ssh-askpass-sublime"
```

---

Hi @danieldk, thanks for the suggestion. I wasn't aware that makeWrapper had a --argv0 argument, that is useful. The generated ssh-askpass-sublime script is not quite the same as the manually created version as it also causes some other unnecessary command-line arguments to be passed as well due to it adding "$@", but from my testing this doesn't seem to interfere with it working properly so I've updated the branch with this change.

---

Awesome, thanks a lot! I was trying to nix-review this PR, but it seems that the hashes incorrect?

```
hash mismatch in fixed-output derivation '/nix/store/w9n393mak5rgsqzm85j46r0c7rwrcpgy-sublime_merge_build_2011_x64.tar.xz':
  wanted: sha256:0r5qqappaiicc4srk08az2vx42m7b6a75yn2ji5pv4w4085hlrzp
  got:    sha256:0hi9glnvkv46hdzp70dglrc3pj9zlrmcfn4bm6r86hiiragibm89
```

It seems that the same hash is used for the dev and non-dev version?

---

Could you add another commit to this PR that fixes the hash(es)?

---

Huh, good catch, looks like this was introduced in the last version bump to the package, I don't usually use the dev version so I didn't notice. Just pushed up another commit to this branch that fixes this.